encodings/sparse/src/canonical/builder.rs

```diff
@@ -0,0 +1,155 @@
+use std::mem::MaybeUninit;
+
+use vortex_array::builders::{BoolBuilder, PrimitiveBuilder};
+use vortex_array::variants::PrimitiveArrayTrait;
+use vortex_array::IntoArrayVariant;
+use vortex_dtype::{match_each_integer_ptype, NativePType};
+use vortex_error::{VortexError, VortexExpect, VortexResult};
+use vortex_scalar::PValue;
+
+use crate::SparseArray;
+
+#[allow(clippy::cast_possible_truncation)]
+pub(super) fn canonicalize_primitive_into<T: NativePType + TryFrom<PValue, Error = VortexError>>(
+    sparse: &SparseArray,
+    builder: &mut PrimitiveBuilder<T>,
+) -> VortexResult<()> {
+    // Fill the output buffer with the fill value.
+    let fill_value: T = sparse
+        .fill_scalar()
+        .as_primitive()
+        .typed_value()
+        .vortex_expect("fill value");
+
+    builder
+        .values
+        .spare_capacity_mut()
+        .fill(MaybeUninit::new(fill_value));
```

Comment:
Will this not fill past the arr len? Is that performance or correct?

Reply:
Ah I see, I can just trim this to `[..len]` and that's probably better